281 change meeting location to real addresses

DanR2003 opened the pull request:
PR Description: Enhanced Meeting Locations and Map Integration 🌍📍

    Meeting Locations:
    Introduced a Location class to support real, mappable locations for meetings.
    Added functionality to display user meetings on the map.

    Search and Suggestions:
        Created LocationRepository interface and NominatimLocationRepository for location searching and parsing.
        Developed LocationViewModel to manage location suggestions.

    Location Field Component:
        Added LocationField function in SharedComponents for dropdown location suggestions.
        Integrated it into AddMeeting and EditMeeting screens.

    Map Enhancements:
        Updated the map type to satellite view for better visualization.

    Testing:
        Unit tests for LocationViewModel and NominatimLocationRepository.
        UI test for the LocationField dropdown menu.

 This update improves location-based interactions and visualization, making meeting location management more intuitive and engaging and also enabling meeting places to be shown on the map so friends can be shown their way to them.

ifaresse reviewed (changes requested):
The dropdown list for recommended location disappear after 1sec if we write the location quickly.
![photo_2024-12-11_22-14-00](https://github.com/user-attachments/assets/8314b92a-e51f-4d69-bc26-4950c3c14b96)
The satellite map looks good however the name of the friends on it with the current implementation is hardly readable maybe go back to the previous version like the one currently for the meeting.
![photo_2024-12-11_21-59-31](https://github.com/user-attachments/assets/356324c9-e2ed-4089-b60c-6340a5ca1ed1)
The meeting displays well on the map, maybe indicate the city in the dropdown list because I wanted to the select the Grimper.ch near Lausanne.

DanR2003 commented on app/src/main/java/com/swent/suddenbump/model/meeting_location/NominatimLocationRepository.kt:
I know we are using google geocoding in the overview, but for my feature (meeting location suggestions), I just felt Nominatim might be easier to implement and more straight-ford, so that's why i picked it for this task.

DanR2003 commented on app/src/main/java/com/swent/suddenbump/model/meeting_location/Location.kt:
```diff
@@ -0,0 +1,3 @@
+package com.swent.suddenbump.model.meeting_location
+
+data class Location(val latitude: Double = 0.0, val longitude: Double = 0.0, val name: String = "")
```
I get what you're saying, but after all I think it may be better to stick with this name. The confusion is actually gone when you check the package of the class: it's Location inside LocationMeeting. I also feel like giving a longer, perhaps more suggestive name might make the code harder to read.

DanR2003 commented:
> The dropdown list for recommended location disappear after 1sec if we write the location quickly. ![photo_2024-12-11_22-14-00](https://private-user-images.githubusercontent.com/73244226/394919720-8314b92a-e51f-4d69-bc26-4950c3c14b96.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwMjExMjEsIm5iZiI6MTczNDAyMDgyMSwicGF0aCI6Ii83MzI0NDIyNi8zOTQ5MTk3MjAtODMxNGI5MmEtZTUxZi00ZDY5LWJjMjYtNDk1MGMzYzE0Yjk2LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDE2MjcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2ZDZhMzY5OWY2OTZmNzA3MmU0ZGJiNTZhNmRhMmJjNmQ0ODNlMDIwYjU5YzA0M2FmMzIzNTdiMGE1MjhhMDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WQQRgzf59ffKPypXSLiPXwHyryjtek3qMg_WZtmKobg) The satellite map looks good however the name of the friends on it with the current implementation is hardly readable maybe go back to the previous version like the one currently for the meeting. ![photo_2024-12-11_21-59-31](https://private-user-images.githubusercontent.com/73244226/394916225-356324c9-e2ed-4089-b60c-6340a5ca1ed1.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwMjExMjEsIm5iZiI6MTczNDAyMDgyMSwicGF0aCI6Ii83MzI0NDIyNi8zOTQ5MTYyMjUtMzU2MzI0YzktZTJlZC00MDg5LWI2MGMtNjM0MGE1Y2ExZWQxLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDE2MjcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3MThhNzFjMTVmYzYxNTgxM2ZmMjViMWM0ZjQzNzY1MGUyNzY1ZWE2ODgyYmZjNDljYmZjYzRiMjM1M2RiMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.r2fevOJH2o6YsxraPfQqqspwd4jokss4csl0gDluPoA) The meeting displays well on the map, maybe indicate the city in the dropdown list because I wanted to the select the Grimper.ch near Lausanne.

I fixed the friend's name on the map: now they are white and bolded so they can be easily read.
Regarding the dropdown list for recommended location disappearing after 1sec if we write the location quickly, I only get this issue when there are no suggestions for what i just quickly wrote.
I just added a "More" button to show more locations if the number of suggested location is more than 3. However, if you search for a location that is not found at all by Nominatim, it still won't show unfortunately. I also added additional checks for location submission (has to be a real location).

ifaresse reviewed (approved):
The new friends name looks good !
![image](https://github.com/user-attachments/assets/d34a8150-e473-46d2-817c-e28764e135b2)
However the Nominatim feature with the dropdown list is broken on my end, it was complicated to reach the "more..." button this one works but for some location that are available on Nominatim sometimes it doesn't show the list at all or it instantly disapears.
I approve it because you won't have time to fix it this sprint but we need to look into it next sprint.

DanR2003 commented:
> The new friends name looks good ! ![image](https://private-user-images.githubusercontent.com/73244226/395302721-d34a8150-e473-46d2-817c-e28764e135b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwNDA2OTYsIm5iZiI6MTczNDA0MDM5NiwicGF0aCI6Ii83MzI0NDIyNi8zOTUzMDI3MjEtZDM0YTgxNTAtZTQ3My00NmQyLTgxN2MtZTI4NzY0ZTEzNWIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDIxNTMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0ZTExNTY2NWNhYTU2ZjE5ZWZkODNkNzJmZDIzNmNmOTJiNWNjMjBkZGRlMzI3Zjc3MDk1NTcxYjA3Y2Y0N2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nyyI18H5STUD-mgl2pI6wWezoyfTg9aI__fZB2wOwmY) However the Nominatim feature with the dropdown list is a little glitchy on my end but I think it's because of Nominatim as the implementation is looking good and similar to the bootcamp, it was complicated to reach the "more..." button this one works but for some location that are available on Nominatim sometimes it doesn't show the list at all or it instantly disapears.

These glitches may be down to how Nominatium responds to our queries. On my end, I don't seem do encounter them too much, but we may look into this next  sprint and actually try and see if using GeoCoding would be any faster or better
